homeassistant/components/surepetcare/__init__.py

```diff
@@ -62,6 +66,13 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+LOCK_STATE_SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_FLAP_ID): cv.positive_int,
+        vol.Required(CONF_LOCK_STATE): cv.string,
```

Comment:
If the lock state parameter only has specific allowed values, we should validate those with `vol.In` besides `cv.string`. Use `vol.All` to wrap both validators.

Reply:
Validated that provided lock state is one of the known states